packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -1081,21 +1101,40 @@ export class DatabaseCluster extends DatabaseClusterNew {
   constructor(scope: Construct, id: string, props: DatabaseClusterProps) {
     super(scope, id, props);
 
-    const credentials = renderCredentials(this, props.engine, props.credentials);
-    const secret = credentials.secret;
+    let cluster: CfnDBCluster;
+    if (!props.manageMasterUserPassword) {
+      const credentials = renderCredentials(this, props.engine, props.credentials);
+      const secret = credentials.secret;
 
-    const cluster = new CfnDBCluster(this, 'Resource', {
-      ...this.newCfnProps,
-      // Admin
-      masterUsername: credentials.username,
-      masterUserPassword: credentials.password?.unsafeUnwrap(),
-    });
+      cluster = new CfnDBCluster(this, 'Resource', {
+        ...this.newCfnProps,
+        // Admin
+        masterUsername: credentials.username,
+        masterUserPassword: credentials.password?.unsafeUnwrap(),
+      });
 
-    this.clusterIdentifier = cluster.ref;
-    this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;
+      this.clusterIdentifier = cluster.ref;
+      this.clusterResourceIdentifier = cluster.attrDbClusterResourceId;
+
+      if (secret) {
+        this.secret = secret.attach(this);
+      }
+    } else {
+      cluster = new CfnDBCluster(this, 'Resource', {
+        ...this.newCfnProps,
+        // Admin
+        // Don't set password if RDS is managing credentials.
```

Comment:
Does the L2 DatabaseCluster construct support non-Aurora engines? The [DatabaseClusterEngine](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseClusterEngine.html) appears to only support Aurora, so I don't think that the multi-AZ check is required. I'm also sure that `manageMasterUserPassword` works on a RDS cluster with a single instance when the db engine is Aurora based.

Reply:
The non Aurora `IClusterEngine` variants _are_ supported by the construct, but the `mysql` and `postgres` engines are both "Multi-AZ DB clusters". As far as I can tell, the only way to deploy those two engines _not_ as "Multi-AZ DB clusters" is to use `DatabaseInstance` instead of this construct.

Reply:
@a-bigelow Thanks for the clarification! Based on that information I believe that a specific set of checks for a multi-AZ cluster wouldn't be necessary.

Tagging @lpizzinidev as conversation starter.